pkg/agent/runner.go

```diff
@@ -289,22 +289,35 @@ func (r *Runner) State(ctx context.Context) (*RunnerState, error) {
 
 func (r *Runner) Spawn(ctx context.Context, vmInfoUpdated util.CondChannelReceiver) {
 	go func() {
+		// Trigger restart if necessary *after* we've handled any panics.
+		defer r.global.TriggerRestartIfNecessary(ctx, r.vm.NamespacedName(), r.podIP)
```

Comment:
I'd just put this on like 305 within the same defer statement: same effect and you don't have to think about LIFO defers (which are great, but also confusing.

Reply:
Done